src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -246,6 +274,130 @@ export class InstructorCoursesPageComponent implements OnInit {
+      modalRef.componentInstance.selectedFeedbackSessions = new Set(response.feedbackSessions);
+      modalRef.result.then((result: CopyCourseModalResult) => this.createCourse(result), () => {});
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessageService.showErrorToast(resp.error.message);
+    });
+  }
+
+  /**
+   * Creates a new course with the selected feedback sessions
+   */
+  createCourse(result: CopyCourseModalResult): void {
+    this.isCopyingCourse = true;
+    this.numberOfSessionsCopied = 0;
+    this.totalNumberOfSessionsToCopy = result.totalNumberOfSessions;
+    this.copyProgressPercentage = 0;
+
+    this.courseService.createCourse({
+      courseName: result.newCourseName,
+      timeZone: result.newTimeZone,
+      courseId: result.newCourseId,
+    })
+    .subscribe(() => {
+      // Wrap in a Promise to wait for all feedback sessions to be copied
+      const promise: Promise<void> = new Promise<void>((resolve: () => void, _reject: () => void) => {
+        result.selectedFeedbackSessionList.forEach((session: FeedbackSession) => {
+          this.copyFeedbackSession(session, result.newCourseId, result.oldCourseId)
+            .pipe(finalize(() => {
+              this.numberOfSessionsCopied += 1;
+              this.copyProgressPercentage =
+                Math.round(100 * this.numberOfSessionsCopied / this.totalNumberOfSessionsToCopy);
+              this.progressBarService.updateProgress(this.copyProgressPercentage);
+              if (this.numberOfSessionsCopied === this.totalNumberOfSessionsToCopy) {
+                resolve();
+              }
+            }))
+            .subscribe();
+        });
+      });
+
+      promise.then(() => {
+        this.courseService
+          .getCourseAsInstructor(result.newCourseId)
+          .subscribe((course: Course) => {
+            this.activeCourses.push(this.getCourseModelFromCourse(course));
+            this.activeCoursesList.push(course);
+            this.allCoursesList.push(course);
+            this.activeCoursesDefaultSort();
+            this.isCopyingCourse = false;
+            this.statusMessageService.showSuccessToast('The course has been added.');
+          });
+      });
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessageService.showErrorToast(resp.error.message);
+      this.isCopyingCourse = false;
+      this.hasLoadingFailed = true;
+    });
+  }
+
+  /**
+   * Gets a CourseModel from courseID
+   */
+  private getCourseModelFromCourse(course: Course): CourseModel {
+    let canModifyCourse: boolean = false;
+    let canModifyStudent: boolean = false;
+    if (course.privileges) {
+      canModifyCourse = course.privileges.canModifyCourse;
+      canModifyStudent = course.privileges.canModifyStudent;
+    }
+    const isLoadingCourseStats: boolean = false;
+    const courseModel: CourseModel = { course, canModifyCourse, canModifyStudent, isLoadingCourseStats };
+    return courseModel;
```

Comment:
`courseModel` needs not be a variable? Can just return the previous line directly.